ale/base/data_isis.py

```diff
@@ -36,187 +37,144 @@ def read_table_data(table_label, cube):
-    if all (key in field_data for key in ('AV1','AV2','AV3')):
-        results['AngularVelocities'] = np.array( [ [av1, av2, av3] for av1, av2, av3 in zip(field_data['AV1'],field_data['AV2'],field_data['AV3']) ] )
-    if 'ET' in field_data:
-        results['Times'] = np.array(field_data['ET'])
-    if all (key in field_data for key in ('J2000Ang1','J2000Ang2','J2000Ang3')):
-        results['EulerCoefficients'] = np.array([field_data['J2000Ang1'],field_data['J2000Ang2'],field_data['J2000Ang3']])
-        results['BaseTime'] = field_data['J2000Ang1'][-1]
-        results['TimeScale'] = field_data['J2000Ang2'][-1]
-
-    if 'TimeDependentFrames' in label:
-        results['TimeDependentFrames'] = np.array(label['TimeDependentFrames'])
-    if 'CkTableOriginalSize' in label:
-        results['CkTableOriginalSize'] = label['CkTableOriginalSize']
-    if all (key in label for key in ('ConstantRotation','ConstantFrames')):
-        const_rotation_mat = np.array(label['ConstantRotation'])
-        results['ConstantRotation'] = np.reshape(const_rotation_mat, (3, 3))
-        results['ConstantFrames'] = np.array(label['ConstantFrames'])
-    if all (key in label for key in ('PoleRa','PoleDec','PrimeMeridian')):
-        results['BodyRotationCoefficients'] = np.array( [label['PoleRa'],label['PoleDec'],label['PrimeMeridian']] )
-    if all (key in label for key in ('PoleRaNutPrec','PoleDecNutPrec','PmNutPrec','SysNutPrec0','SysNutPrec1')):
-        results['SatelliteNutationPrecessionCoefficients'] = np.array( [label['PoleRaNutPrec'],label['PoleDecNutPrec'],label['PmNutPrec']] )
-        results['PlanetNutationPrecessionAngleCoefficients'] = np.array( [label['SysNutPrec0'],label['SysNutPrec1']] )
-
-    return results
-
-def parse_position_table(label, field_data):
+    # Case 1, the table has positions at discrete times
+    if 'J2000X' in table:
+        ephemeris_times = table['ET']
+        positions = np.array([table['J2000X'],
+                              table['J2000Y'],
+                              table['J2000Z']]).T
+        rotated_pos = rotation.apply_at(positions, ephemeris_times)
+    # Case 2, the table has coefficients of polynomials for the position
+    elif 'J2000SVX' in table:
+        ephemeris_times = np.linspace(table['SpkTableStartTime'],
+                                      table['SpkTableEndTime'],
+                                      table['SpkTableOriginalSize'])
+        base_time = table['J2000SVX'][-1]
+        time_scale = table['J2000SVY'][-1]
+        scaled_times = (ephemeris_times - base_time) / time_scale
+        coeffs = np.array([table['J2000SVX'][:-1],
+                           table['J2000SVY'][:-1],
+                           table['J2000SVZ'][:-1]]).T
+        positions = polyval(scaled_times, coeffs).T
+        rotated_pos = rotation.apply_at(positions, ephemeris_times)
+    else:
+        raise ValueError('No positions are available in the input table.')
+    return rotated_pos, ephemeris_times
```

Comment:
The change to this function seems pretty significant, has the rotation table already been parsed and we just need to actually rotate it now?

Reply:
This is an entirely new function. Git just though it was changed. I completely removed the parse_rotation_table and parse_position_table methods. Now, there's just one parse_table method that copies over all of the PVL keywords from the table label and converts the binary fields into lists.